client/src/components/Overview/Overview.jsx

```diff
@@ -19,13 +19,15 @@ function Overview({ productId }) {
           axios.get(`/products/${productId}/styles`),
         ]);
 
-        const sortedStyles = stylesResponse.data.results.sort((style1, style2) => (
-          style1.styled_id - style2.styled_id
-        ));
+        const sortedStyles = stylesResponse.data.results.sort(
+          (style1, style2) => style1.styled_id - style2.styled_id
+        );
 
         setProduct(productResponse.data);
         setStyles(sortedStyles);
-        setSelectedStyleId(sortedStyles.find((style) => style['default?']).style_id);
+        setSelectedStyleId(
```

Comment:
You can initialize state lazily: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Does this provide an advantage over `useEffect` without subscribers? I also need to set `product` and `styles` from data that I pull from an API. But the data is fetched only one the initial render since the subscribers array is empty. I'm now realizing that the useEffect hook should run when `productId` changes though, since the data should be refetched whenever a new product is being displayed.